inst/doc/shared_cache.Rmd

```diff
@@ -1,52 +1,191 @@
+
+  <ind>1_data/out/selected_sites.rds.ind:</ind>
+    command: select_sites(
+      ind_file = target_name,
+      input_ind_file = <ind>'1_data/out/compiled_data.rds.ind'</ind>,
+      gd_config = <data>'lib/cfg/gd_config.yml'</data>)
+</pre>
+
+Functions used in the above code snippet: 
+
+<pre> 
+gather_and_share_stream_data = function(<ind>ind_file</ind>, state, <data>gd_config</data>){
+  
+  <data>temp</data> <- readNWISdata(stateCd = state, parameterCd = '00010', service = 'dv')
+
+  data_file <- as_data_file(<ind>ind_file</ind>) # convert indicator file to data file format (drops .ind suffix)
+  saveRDS(<data>temp</data>, data_file)
+  gd_put(remote_ind = <ind>ind_file</ind>, local_source = <data>data_file</data>, config_file = <data>gd_config</data>)
+}
+
+select_sites = function(<ind>ind_file</ind>, <ind>input_ind_file</ind>, <data>gd_config</data>){
+
+  <data>temp</data> = readRDS(sc_retrieve(<ind>input_ind_file</ind>))
+  
+  <data>temp_sites</data> <- <data>temp</data> %>%
+    dplyr::filter(
+      dateTime > as.POSIXct('2012-01-01')) %>%
+    dplyr::select(site_no, dateTime) 
+    
+  data_file <- as_data_file(<ind>ind_file</ind>)
+  saveRDS(<data>temp_sites</data>, data_file)
+  gd_put(remote_ind = <ind>ind_file</ind>, local_source = <data>data_file</data>, config_file = <data>gd_config</data>)
+}
+
+</pre> 
 
-Disadvantages of a shared cache (as currently implemented):
-* In a fast-paced collaborative development environment (e.g., a 'sprint'), it is challenging to maintain synchrony between the shared cache (the data) and the git repository (the metadata). Asynchrony is not a deal-breaker but does lead to more rebuilding than would be required for a slower-paced project.
-* Though we've done much to ensure this doesn't happen, it's conceivable that metadata will become corrupt relative to the data. Some monitoring and very occasional full rebuilding is recommended when practical.
-* Old files no longer referenced by the code can accumulate on the shared cache unless manually deleted. Though these will not interfere with ongoing analysis, they can take up storage space unnecessarily.
 
 ### Guidelines
 
-Projects using a shared cache should follow these guidelines:
+####Projects using a shared cache should follow these guidelines:
+
+* Use <ind>**indicator files**</ind> (usually an `.ind` suffix) to represent most or all of the chain of connected <target>**targets**</target> within your remake files. Each `.ind` file should be one of two products of a recipe (e.g. R function call), where the other product is the creation of a <data>**data file**</data>, either locally and/or in the shared cache. The <ind>**indicator files**</ind> are the only products that are declared to `remake`, while the <data>**data files**</data> remain hidden. This allows the <ind>**indicator files**</ind> to indirectly represent the <data>**data files**</data> to `remake` thereby enabling compatibility between `remake` and the shared cache. 
```

Comment:
I would suggest adding an example of what doesn't need to have an `ind` file. Such as things that are quick to build (like reading info from a yaml file for state names or something), maybe targets that require < 5 seconds of build time.

Reply:
Yeah, when building is likely to be faster than downloading

Reply:
good point, I will add that